src/include/enclave/lthread_int.h

```diff
@@ -71,4 +71,20 @@ static inline uint64_t _lthread_timespec_to_usec(const struct timespec* ts)
     return (ts->tv_sec * 1000000) + ts->tv_nsec / 1000;
 }
 
+// TODO: should this be static?
+static inline struct schedctx *__scheduler_self()
+{
+	struct schedctx *self;
+	__asm__ __volatile__ ("mov %%gs:48,%0" : "=r" (self) );
+	return self;
+}
+
+
+static inline struct lthread_sched*
+lthread_get_sched()
+{
+    struct schedctx *c = __scheduler_self();
```

Comment:
Are there noe any other callers of `__scheduler_self`?  If not, it can be inlined here.

Reply:
Its also used to setup the `schedctx` struct in `enclave_oe.c:init_ethread_tp`.